audio_service/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java

```diff
@@ -58,22 +61,52 @@
  */
 public class AudioServicePlugin implements FlutterPlugin, ActivityAware {
     private static String flutterEngineId = "audio_service_engine";
-    /** Must be called BEFORE any FlutterEngine is created. e.g. in Application class. */
+
+    /**
+     * Must be called BEFORE any FlutterEngine is created. e.g. in Application class.
+     */
     public static void setFlutterEngineId(String id) {
         flutterEngineId = id;
     }
+
     public static String getFlutterEngineId() {
         return flutterEngineId;
     }
+
     public static synchronized FlutterEngine getFlutterEngine(Context context) {
         FlutterEngine flutterEngine = FlutterEngineCache.getInstance().get(flutterEngineId);
+
+        // Since flutter 3 offer disabling SKParagraph
+        // And it has a bug that effected most of android flutter apps
+        // Check for the flag if it's available in manifest file
+
+        ApplicationInfo applicationInfo =
+                null;
+        String[] args = new String[1];
+        try {
+            applicationInfo = context
+                    .getPackageManager()
+                    .getApplicationInfo(
+                            context.getPackageName(), PackageManager.GET_META_DATA);
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        if (applicationInfo != null) {
+            Bundle metaData = applicationInfo.metaData;
+            if (metaData != null) {
+                final String skEnabled = metaData.getBoolean("io.flutter.embedding.android.EnableSkParagraph", true) ? "true" : "false";
+                args[0] = ("--enable-skparagraph=".concat(skEnabled));
+            }
+        }
+
         if (flutterEngine == null) {
             // XXX: The constructor triggers onAttachedToEngine so this variable doesn't help us.
             // Maybe need a boolean flag to tell us we're currently loading the main flutter engine.
-            flutterEngine = new FlutterEngine(context.getApplicationContext());
+            flutterEngine = new FlutterEngine(context.getApplicationContext(), args);
             String initialRoute = null;
             if (context instanceof FlutterActivity) {
-                final FlutterActivity activity = (FlutterActivity)context;
+                final FlutterActivity activity = (FlutterActivity) context;
```

Comment:
Try to ensure that the diff only shows lines that you have actually changed. This one only changes formatting which is to be avoided.